File: Uva/10344.cpp
#include <bits/stdc++.h>
using namespace std;
typedef vector<int> vec;
bool check;
void each(vec &arr, int i, int value)
{
    if (i > 4 or check == true)
    {
        if (value == 23)
        {
            check = true;
        }
        return;
    }
    if (value > 250 and value < -250)
        return;
    each(arr, i + 1, value + arr[i]);
    each(arr, i + 1, value - arr[i]);
    each(arr, i + 1, value * arr[i]);
}
void permu(vec &arr)
{
    do
    {
        each(arr, 1, arr[0]);
    } while (next_permutation(arr.begin(), arr.end()));
}
int main()
{
    // freopen("input.txt", "r", stdin);
    vector<int> arr(5);
    while (scanf("%d", &arr[0]) and arr[0])
    {
        for (int i = 1; i < 5; i++)
            scanf("%d", &arr[i]);
        check = false;
        sort(arr.begin(), arr.end());
        permu(arr);
        puts(check ? "Possible" : "Impossible");
    }
}

File: Uva/10344.py
check = True


def each(arr, i, value):
    global check
    if(i > 4 or check == True):
        if(value == 23):
            check = True
        return
    each(arr, i+1, value+arr[i])
    each(arr, i+1, value-arr[i])
    each(arr, i+1, value*arr[i])


def perm(a, k=0):
    global check
    if k == len(a) or check == True:
        if(check == True):
            return
        each(a, 1, a[0])
        if(check == True):
            return
    else:
        for i in range(k, len(a)):
            a[k], a[i] = a[i], a[k]
            perm(a, k+1)
            if(check == True):
                return
            a[k], a[i] = a[i], a[k]


arr = [int(x) for x in input().split(" ")]
while arr[0] != 0:
    check = False
    perm(arr)
    print("Possible") if check else print("Impossible")
    arr = [int(x) for x in input().split(" ")]